### ISSUE TITLE
Suspect numbers in Airloop and Facility component loads reports

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6337]
Issue overview
--------------
Using RefBldgFullServiceRestaurantNew2004_Chicago with     ZoneComponentLoadSummary, AirLoopComponentLoadSummary, and FacilityComponentLoadSummary.  Comparing the zone, system and facility peak load components, something doesn't add up.

For example, both zones, both airloops, and the facility all report the same time of cooling peak as 07/21 06:00 but the reported outdoor conditions, and other loads like People don't match.

Fenestration conduction seems to go from positive to negative from zone to system to facility.  Wondering if the subtraction of adjacent surface radiant in OutputReportTabular::GetDelaySequences is accumulating every time this is called.  feneCondInstantSeq is handled differently from all of the other components in this function - passing the full 3d array back and forth, where it looks like the others are only passing back 1d arrays based on other untouched data.

The reported date and time of peak should also include the name of the sizing period.  And do these work with a SizingPeriod longer than a day?

Also, I can't find an example file that produces these reports.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by JasonGlazer** [comment]
@larryscheier pointed out some problems using the 5ZoneWarmestVFD.idf file which has five zones and one airloop.

1) In Engineering Checks for both cooling and heating, the values for "Airflow per Total Capacity [m3/s-W]" and "Airflow per Floor Area [m3/s-m2]" can be tiny and the number of digits insufficient to convert to IP units. I suggest increasing the number of significant digits to 8.
   Airflow per Floor Area [m3/s-m2]    0.0008
   Airflow per Total Capacity [m3/s-W]    0.0000

2) (see sheet ComparingFacilityVsAirloop) I would have expected the AirLoop and Facility level load component reports to be identical since there is a single airloop but:

 >>Estimated Cooling Peak Load Components
    Fenestration Conduction: Facility =  4432.19 vs Airloop = 3282.69

 >> Cooling Peak Conditions
   Supply Air Temperature [C]    Facility =     0   vs Airloop =     12.8
   Mixed Air Temperature [C]    Facility =         vs Airloop =     26.61
   Main Fan Air Flow [m3/s]    Facility =     1.29    vs Airloop =     1.68
   Peak Sensible Load with Sizing Factor [W]    Facility =     16253.79    vs Airloop =     22301.42
   Difference Due to Sizing Factor [W]    Facility =     3750.87    vs Airloop =     0
   Peak Sensible Load [W]    Facility =     12502.91    vs Airloop =     22301.42
   Estimated Instant + Delayed Sensible Load [W]    Facility =     13826.38    vs Airloop =     12676.87
   Difference Between Peak and Estimated Sensible Load [W]    Facility =     -1323.5    vs Airloop =     9624.55

 >>Estimated Heating Peak Load Components                        
    Interzone Ceiling    Facility =     -4494.3    Airloop =     Interzone Ceiling    -4212.6
    Exterior Wall    Facility =     -2222.7    Airloop =     Exterior Wall    -2180.9
    Interzone Wall    Facility =     -177.1    Airloop =     Interzone Wall    75.09
    Fenestration Conduction    Facility =     -13991.6    Airloop =     Fenestration Conduction    -10315.1

 >> Heating Peak Conditions
   Supply Air Temperature [C]    Facility =     0    vs airloop =     16.7
   Mixed Air Temperature [C]    Facility =         vs airloop =     -10.1
   Main Fan Air Flow [m3/s]    Facility =     0.39    vs airloop =     1.68
   Peak Sensible Load with Sizing Factor [W]    Facility =     -14119.9    vs airloop =     -15578.3
   Difference Due to Sizing Factor [W]    Facility =     -3258.4    vs airloop =     0
   Peak Sensible Load [W]    Facility =     -10861.4    vs airloop =     -15578.3
   Estimated Instant + Delayed Sensible Load [W]    Facility =     -18166.7    vs airloop =     -14522
   Difference Between Peak and Estimated Sensible Load [W]    Facility =     7305.31    vs airloop =     -1056.3

3) (see sheet ComparingSumOfZnPktoBlkAirloop) With Cooling Peak Conditions section, I tried comparing the sum-of-zones to airloop. I would expect the total values for sum-of-zones to be larger than the associate airloop (coincident) values but: (my guess is the Zone sizing factors were not aways applied properly at the Airloop level perhaps?)

>>Cooling Peak Conditions
    Main Fan Air Flow [m3/s]    Sum-of-zones =     1.29    Airloop =     1.68
    Peak Sensible Load with Sizing Factor [W]    Sum-of-zones =     16253.79    Airloop =     22301.42
    Difference Due to Sizing Factor [W]    Sum-of-zones =     3750.88    Airloop =     0
    Peak Sensible Load [W]    Sum-of-zones =     12502.92    Airloop =     22301.42
  The good news :
    Estimated Instant + Delayed Sensible Load [W]    Sum-of-zones =     13104.46    Airloop =     Estimated Instant + Delayed Sensible Load [W]    12676.87

4) (see sheet ComparingSumOfZnPktoBlkAirloop) With Heating Peak Conditions section, I tried comparing the sum-of-zones to airloop. I would expect the total values for sum-of-zones to almost equal the associate airloop (coincident) values but:

>>Heating Peak Conditions                   
    Main Fan Air Flow [m3/s]    Sum-of-zones =     0.39    Airloop =     1.68
    Peak Sensible Load with Sizing Factor [W]    Sum-of-zones =     -14119.8    Airloop =     -15578.3
    Difference Due to Sizing Factor [W]    Sum-of-zones =     -3258.5    Airloop =     0
    Peak Sensible Load [W]    Sum-of-zones =     -10861.4    Airloop =     -15578.3
    Estimated Instant + Delayed Sensible Load [W]    Sum-of-zones =     -10813.9    Airloop =     -14522

**Comment by larryscheier** [comment]
For what it's worth: Here is the associated spreadsheet for the above comparisons: [5ZoneWarmestVFDTable.xlsx](https://github.com/NREL/EnergyPlus/files/1668414/5ZoneWarmestVFDTable.xlsx)

It includes a sheet for the original summary table, another sheet for comparing the Airloop to Facility and a third sheet comparing Sum-of-Zone Components to Airloop Components.

**Comment by EnergyArchmage** [comment]
I have been looking into time of day reporting in the airloop component loads summary table and I think I have found a problem.  It appears that the airloop summary table is filling the text for date and time with values that originally come from the variables CoolPeakDateHrMin and HeatPeakDateHrMin. But these are filled with peaks that are based on zone sensible peak time.  It appears that the value that shows up for airloop table is simply the space sensible peak for the last zone on the air handler.  I believe that the underlying data are for the correct system sizing peak time, it is just the reported text for the time that is wrong.

I suggest to add an argument to CollectPeakZoneConditions for the design day number, and then generating text for date and time on the fly from the design day number and the timestep into day integer on the fly.   There is a useful function in the new coil report table contribution here

https://github.com/NREL/EnergyPlus/blob/Coil-Report-Table/src/EnergyPlus/ReportCoilSelection.cc#L1601

and with that the text date and time generation can look like this:
compLoad.peakDateHrMin = General::TrimSigDigits( WeatherManager::DesDayInput( desDaySelected ).Month ) + "/" + General::TrimSigDigits( WeatherManager::DesDayInput( desDaySelected ).DayOfMonth ) + " "  + coilSelectionReportObj->getTimeText( timeOfMax );

**Comment by larryscheier** [comment]
FYI: I created a simple one-zone airloop and forced the ventilation to zero so the airloop coil would peak at the same time as the zone sensible peak. The "AirLoop Component Load Summary" should exactly match the "Zone Component Load Summary" report but does not for either cooling or heating. See these files: 
[AirloopChecksum_OneZone.zip](https://github.com/NREL/EnergyPlus/files/1686348/AirloopChecksum_OneZone.zip)

**Comment by JasonGlazer** [comment]
Thanks @larryscheier it is even more perplexing that is is having problems with a one-zone airloop case.

**Comment by larryscheier** [comment]
When I run the one-zone airloop case, the component tables match but not the Cooling Peak Conditions:

Peak Sensible Load with Sizing Factor [W] = Zone 8854.88 vs Airloop 9231.26
Difference Due to Sizing Factor [W] = Zone 1154.98 vs Airloop 0
Peak Sensible Load [W] = Zone 7699.89 vs Airloop 9231.26
Estimated Instant + Delayed Sensible Load [W] = Zone 7816.03 vs Airloop 7816.03  =>>OK
Difference Between Peak and Estimated Sensible Load [W] = Zone -116.1 vs Airloop 1415.23

FYI: I ran AirloopChecksum_OneZone_v8.8.idf in the Trane customized version of E+ after @EnergyArchmage had ported over your fixes. If you are not seeing these differences, please send me your AirloopChecksum_OneZone_v8.8Table.csv results.

**Comment by JasonGlazer** [comment]
The Airloop numbers for that subtable come from the FinalSysSizing and CalcSysSizing arrays which are different than the Zone level numbers that come from CalcFinalSizing arrays. I'm no sizing expert but the fact that they are different is expected.

**Comment by larryscheier** [comment]
This is the one zone example with no ventilation which meant that the zone and airloop peaked at the same timestep which is why I expected these values to match. I'm not familiar with what's in the arryas you mention but I suspect that the zone Peak Sensible Load is different than the reported airloop Peak Sensible Load load because the airloop value represents the coil sensible whereas the zone Peak Sensible Load represents the space sensible. I'm not sure why the effect of the Sizing:Parameters object would be ignored by the airloop values. Was that intended?

**Comment by JasonGlazer** [comment]
I'm taking the design peak load from the FinalSysSizing array and the peak design sensible load from the CalcSysSizing array which I would have expected to include the impact of Sizing:Parameters. 

@EnergyArchmage can you provide some advice if these are pulling numbers from the proper places?

**Comment by mjwitte** [comment]
Here are some clues from the eio outuput.
```
 System Sizing Information, SYS0, Cooling, Total, 9762.47, 0.62106, 0.62106, ANNUAL COOLING DESIGN CONDITION10, 11/21 07:15:00
 System Sizing Information, SYS0, Heating, Sensible, 5.40E-004, 0.99415, 0.99415, ANNUAL COOLING DESIGN CONDITION11, 12/21 07:45:00
```
Looking at heating, it's picking the system peak heating from a cooling design day and the system peak heating is basically zero.
Looking at cooling, the system component loads is reporting 9762.47 as the peak sensible, but this says it's Total.
So, why is the system peak heating zero? Because this file has a cold heating supply temp. And the heating system max airflow ratio is 0.3. Not sure if it's possible to pick up the reheat loads to get a true system heating peak.
```
  Sizing:System,
    SYS0,                    !- AirLoop Name
    Total,                   !- Type of Load to Size On
    autosize,                !- Design Outdoor Air Flow Rate {m3/s}
    0.3,                     !- Central Heating Maximum System Air Flow Ratio
    12.7777777777778,        !- Preheat Design Temperature {C}
    0.000658960788112938,    !- Preheat Design Humidity Ratio {kgWater/kgDryAir}
    12.7777777777778,        !- Precool Design Temperature {C}
    0.00624385906033234,     !- Precool Design Humidity Ratio {kgWater/kgDryAir}
    12.2964040076683,        !- Central Cooling Design Supply Air Temperature {C}
    12.2964040076683,        !- Central Heating Design Supply Air Temperature {C}
```

**Comment by mjwitte** [comment]
Forget what I said about the peak cooling loads. I was looking at the old output (before this fix) for that. But the heating comments still stand.

Per ssz, peak system sensible is 9231 and total is 9762 and the occur on design period 11 (which is design day 10 in November) at 13:00. So that agrees.
Comparing the time series in ssz vs zsz, the max delta is 1531W which is close to the peak fan power:
```
 Component Sizing Information, Fan:SystemModel, SYS0SF-1, Design Size Design Maximum Air Flow Rate [m3/s], 0.99415
 Component Sizing Information, Fan:SystemModel, SYS0SF-1, Design Electric Power Consumption [W], 1391.86643
```
But zsz is before sizing factor (it's peak load is 8854). The ssz mass flow rates are with sizing factor (comparing to zsz they all differ by 1.15x). I wonder if the ssz cool cap is with sizing factor?

**Comment by mjwitte** [comment]
No, ssz cool caps are all recalculated in UpdateSysSizing using the mass flow rates with sizing factor. And fan heat is not added here, it's added to the central cooling coil design load, but doesn't show in the system design load.

**Comment by larryscheier** [comment]
I doubt what we need is available in FinalSysSizing and CalcSysSizing arrays. From my perspective:
Airloop Peak Sensible Load = Sum of (Zone Peak Sensible Load for attached zones at time of Airloop coil peak)
Airloop Peak Sensible Load with Sizing Factor = Sum of (Zone Sizing Factor *Zone Peak Sensible Load for attached zones at time of Airloop coil peak)

**Comment by larryscheier** [comment]
So based on today's sizing call, here are my suggestions for the Cooling Peak Conditions (and Heating Peak Conditions) sections of the Airloop load component reports:

1: Time of Peak (and associated load components) should be based on when the associated airloop coil peaks, i.e., if Sizing:System "Type of Load to Size On" field is Sensible, then the pull airloop components from time of coil sensible peak, if Total then pull airloop components from time of coil total peak, if VentilationRequirement pull from that coil peak. This peak time is known from the newly added coil selection code (or perhaps elsewhere), but if you want to leave as is, i.e., based on sensible coil load, that's fine.

2: The following rows seem fine as is:

Outside Dry Bulb Temperature [C]
Outside Wet Bulb Temperature [C]
Outside Humidity Ratio at Peak   [kgWater/kgAir]
Supply Air Temperature [C]
Mixed Air Temperature [C]
Supply Air Temperature [C]
Mixed Air Temperature [C]
Main Fan Air Flow [m3/s]
Outside Air Flow [m3/s]
Estimated Instant + Delayed Sensible Load [W]

There was some debate about whether or not the displayed Main Fan Air Flow should be adjusted when heating airflow is larger, but let's keep as is, so the airloop's mixed air condition closely reflects entering coil condition as possible. 

3: If possible, these fields should be weighted by each zone's terminal airflow (or leave as is if that's too painful):
Zone Dry Bulb Temperature [C]
Zone Relative Humidity [%]
Zone Humidity Ratio at Peak   [kgWater/kgAir]

4: These rows should be based on same method as used to populate the same rows in the Facility Load Component report:
Peak Sensible Load with Sizing Factor [W]
Difference Due to Sizing Factor [W]
Peak Sensible Load [W]
Difference Between Peak and Estimated Sensible Load [W]

Let me know if more clarification is needed.

**Comment by mjwitte** [comment]
@larryscheier The method use for Facility Load Component will only gather zone loads - no ventilation loads. Is that OK? If so, the row labels may need to change to reflect that.

> 4: These rows should be based on same method as used to populate the same rows in the Facility Load Component report:
Peak Sensible Load with Sizing Factor [W]
Difference Due to Sizing Factor [W]
Peak Sensible Load [W]
Difference Between Peak and Estimated Sensible Load [W]

@JasonGlazer The info for the flow-weighted average zone temp is calculated for each time step (CoolZoneAvgTempSeq, HeatZoneAvgTempSeq) but not saved in the FinalSysSizing. We could add CoolZoneAvgHumRatSeq and CoolZoneAvgRHSeq and follow the pattern for CoolZoneAvgTempSeq (similar for heating). And we could add ZoneAvgTempAtCoolPeak, ZoneAvgHumRatAtCoolPeak and ZoneAvgRHAtCoolPeak following the pattern of RetTempAtCoolPeak (again similar for heating).

> 3: If possible, these fields should be weighted by each zone's terminal airflow (or leave as is if that's too painful):
Zone Dry Bulb Temperature [C]
Zone Relative Humidity [%]
Zone Humidity Ratio at Peak [kgWater/kgAir] 

Finally, the airloop aggregation needs to be converted to be based on airterminals rather than zones to allow for multiple airloops. I can tackle that but that should probably be a separate PR after this one and #6338 are merged.

**Comment by larryscheier** [comment]
Your question: "The method use for Facility Load Component will only gather zone loads - no ventilation loads. Is that OK?" Yes, that is exactly correct.

**Comment by JasonGlazer** [comment]
@larryscheier concerning item (1), if Sizing:System field "Type of Load to Size On" is set to VentilationRequirement should I use the time of that? Or should I use the time of the Sensible peak? Or the time of the Total peak?

**Comment by larryscheier** [comment]
Use time of VentilationRequirement peak which would be based on total.

**Comment by mjwitte** [comment]
But what exactly does that mean? Peak ventilation flow rate? But that could be constant throughout the day. Or peak coil load? Sensible or Total?

**Comment by larryscheier** [comment]
At the time of the when the airloop cooling coil capacity peaks given that its airflow is based on the VentilationRequirement.

**Comment by larryscheier** [comment]
The peak time would match what @EnergyArchmage displays in his coil selection reports so he mined that from somewhere...

**Comment by JasonGlazer** [comment]
I didn't test it specifically but the SysSizPeakDDNum array has:

SysSizPeakDDNum( ).CoolFlowPeakDD;
SysSizPeakDDNum(  ).TimeStepAtCoolFlowPk( );

Which are supposed to be just that.

**Comment by larryscheier** [comment]
That seems reasonable.

**Comment by EnergyArchmage** [comment]
yeah, I have used this in an alternate set of airloop tables we are producing.  But only if the sizing method is Coincident.  I think it ill defined for non-coincident method. 
	int airflowPeakDDNum = SysSizPeakDDNum( iAirLoop ).CoolFlowPeakDD;
	int airflowPeakTimestep = SysSizPeakDDNum( iAirLoop ).TimeStepAtCoolFlowPk( airflowPeakDDNum );

**Comment by Myoldmopar** [comment]
Closed via #6513
